common/achievement/get.ts

```diff
@@ -0,0 +1,135 @@
+import { prisma } from '../prisma';
+import { User_achievement, achievement_action_type_enum, achievement_type_enum } from '../../graphql/generated';
+import { Achievement, achievement_state } from '../../graphql/types/achievement';
+import { User } from '../user';
+import { ConditionDataAggregations } from './types';
+import { getAchievementState, getCurrentAchievementTemplateWithContext, transformPrismaJson } from './util';
+import { evaluateAchievement } from './evaluate';
+
+// TODO: getAchievementById -> passed user and achievementId to return a single achievement
+// TODO: resolver for nextSteps -> get active sequential achievements and important information
+const getUserAchievements = async (user: User): Promise<Achievement[]> => {
+    const userAchievements = await prisma.user_achievement.findMany({
+        where: { userId: user.userID },
+        include: { template: true },
+    });
+    const userAchievementGroups: { [group: string]: User_achievement[] } = {};
+    userAchievements.forEach((ua) => {
+        if (!userAchievementGroups[ua.template.group]) {
+            userAchievementGroups[ua.template.group] = [];
+        }
+        userAchievementGroups[ua.template.group].push(ua);
+    });
+    const achievements: Achievement[] = await generateReorderedAchievementData(userAchievementGroups, user);
+    return achievements;
+};
+
+const generateReorderedAchievementData = async (groups: { [group: string]: User_achievement[] }, user: User): Promise<Achievement[]> => {
+    const groupKeys = Object.keys(groups);
+    const achievements = await Promise.all(
+        groupKeys.map(async (key) => {
+            const group = groups[key];
+            const sortedGroupAchievements = group.sort((a, b) => a.groupOrder - b.groupOrder);
+            if (sortedGroupAchievements[0].template.type === achievement_type_enum.TIERED) {
```

Comment:
Why do we do this for `TIERED` achievements only?

Reply:
Tiered achievements are the only achievements that are not put together as a group to one achievement for the frontend.
Example:
Sequential achievements use every userAchievement as a step and therefore get processed together to one achievement seen in the frontend.
Streaks just exist as single achievement for a group and do not need this single processing either.
Tiered are steps on a road to higher scores. First you reach 1 then 3 then 5 then 10 ... these achievements all get displayed as a milestone so they are processt one by one.

Reply:
Ahhhh, now I get it.
Could you please add a comment with these details?